lldb/source/Plugins/Process/scripted/ScriptedProcess.cpp

```diff
@@ -288,8 +288,15 @@ Status ScriptedProcess::DoGetMemoryRegionInfo(lldb::addr_t load_addr,
                                               MemoryRegionInfo &region) {
   Status error;
   if (auto region_or_err =
-          GetInterface().GetMemoryRegionContainingAddress(load_addr, error))
+          GetInterface().GetMemoryRegionContainingAddress(load_addr, error)) {
     region = *region_or_err;
+    if (region.GetRange().GetRangeBase() == 0 &&
+        (region.GetRange().GetByteSize() == 0 ||
+         region.GetRange().GetByteSize() == LLDB_INVALID_ADDRESS)) {
```

Comment:
Why would the size be `LLDB_INVALID_ADDRESS`? Did you mean to check that the range base isn't `LLDB_INVALID_ADDRESS`? 

Is this the condition you want?

```
(region.GetRange().GetRangeBase() == 0 || region.GetRange().GetRangeBase() == LLDB_INVALID_ADDRESS) && region.GetRange().GetByteSize() == 0) 
```

Reply:
Last week I had another similar bug where a gdb remote stub was returning start addr 0 size UINT64_MAX as a region response, and I was trying to reject that in addition to addr 0 size 0.  I haven't seen someone send that same incorrect response from a scripted process, though, I should just check for 0/0 here.

Reply:
I'm not wedded to it.  I'm still debating moving both of these checks up in to Process::GetMemoryRegionInfo.  I haven't done it yet for fear of unintended consequences, but any memory region info response that claims 0 bytes or the entire address space is meaningless.

Reply:
A single region that covers the entire address space is fairly boring, but it is the kind of thing I might return as an "I don't know" value (perhaps I'm in some embedded environment where every address really is readable?). What's the problem with that kind of result?

That said, I think this check could be made more robust. `(0, 0)` is the likeliest response, but doesn't this mean that we would still loop if the plugin returns `(0,1)` instead? It sounds to me that if we check that:
- the returned region is of non-zero size
- the returned region actually includes the address being queried (I guess this kinda subsumes the previous item)
then we can guarantee forward progress no matter what the plugin returns (?)

Reply:
> We had a gdb stub returning {0, UINT64_MAX} the other week and it broke IRMemory::FindSpace() which will avoid any memory region with read/write/execute flags if qMemoryRegionInfo packets are supported. The stub claimed the entire address space, FindSpace() said it could not find any address range available, and all expressions broke.

Okay, but what's the alternative? Picking a piece of memory that may overlap with some existing data? It sounds to me like the stub gets exactly what it asked for.

> Yeah, a range of {0, 1} would result in algorithms like FindSpace() looping for a very long time, and be nearly as bad. But so far the two instances I've seen of people return bad ranges are {0,0} and {0,UINT64_MAX}.

True, but if we can change the expression to catch both, why not do it? What I'm suggesting is to change the expression into something like `if (GetRange().GetRangeBase() > addr || GetRange().GetRangeEnd() <= addr)`. The `(0,0) case is subsumed by that, but this also catches any other incorrect response.

Reply:
I see what you mean for checking the address is contained within the range, and rejecting it if not, this works too.

Right now if a stub returns {0, UINT64_MAX} and we don't have a memory allocation packet, IRMemory::FindSpace will find no available virtual address space and all expressions will fail.

My opposition to {0, UINT64_MAX} being accepted is that it's meaningless.  No processor actually has 64-bits of addressing, so telling me that all memory is addressable is obviously wrong.  It will break IRMemoryMap::FindSpace's algorithm, and this response tells us nothing more than if qMemoryRegionInfo was an unsupported packet.

Reply:
I guess the reason I don't like that is that it just feels like a very special case. Like if I send two regions, each covering half the address space, then I'm sort of saying the same thing, but unlike the single region, this response will get accepted. It also doesn't handle the case of a 32 bit address space being full. The fact that this breaks all of expression evaluation might be sort of a good thing actually, as it means the problem is likely to be caught.

If you feel strongly about this, then I don't want to stand in your way, but I would rather not have this special case.

Reply:
No, your points make sense, thanks for sticking with this discussion.  I'll update the patch as you suggest.